Backend/src/main/java/com/example/backend/user/controller/UserController.java

```diff
@@ -26,80 +25,58 @@ public UserController(UserService userService) {
     @PostMapping()
     @ResponseBody
     public ResponseEntity addUser(@RequestHeader String googleTokenEncoded, @RequestBody UserDTO userDTO) {
-        if (userService.existsAny()) {
-            List<AUTHORITY> a = List.of(AUTHORITY.ADMINISTRATOR);
-            String email = JwtVerifier.verifyAndReturnEmail(googleTokenEncoded);
-            if (email == null)
-                return new ResponseEntity("Token is missing or invalid", HttpStatus.UNAUTHORIZED);
-            if (userService.findByEmail(email) == null)
-                return new ResponseEntity("You don't have access to CDB", HttpStatus.UNAUTHORIZED);
-            if (!a.contains(userService.findByEmail(email).getAuthority()))
-                return new ResponseEntity("You don't have premission to this resource", HttpStatus.UNAUTHORIZED);
+        AppUser user = getUser(googleTokenEncoded);
+        try {
+            return new ResponseEntity<>(userService.addUser(userDTO, user), HttpStatus.CREATED);
```

Comment:
Zasto se uvijek userServicu proslijeduje user, zar ne bi bilo onda logicnije da se na controleru odmah pogleda postoji li taj korisnik i koju razinu autorizacije ima i ako mu je dozvoljen pristup tek onda raditi userService.blabla

Reply:
ovo pitanje vrijedi za svugdje gdje se salje user u userservice u controleru*

Reply:
stvar je u tome što je controller zamišljen da samo prosljeđuje podatke koje je dobio u zahtjevu servisu, da ima minimalno posla, zato usera prosljeđujem u servis

Reply:
Slažem se s Hajdukom. Dapače, malo me buni zašto u servisu ne radite ovo AppUser user = getUser(googleTokenEncoded)?